RestaurantReservation.Db/Repositories/OrderItemRepository.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.DataModels;
+using RestaurantReservation.Db.Utilities;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class OrderItemRepository : IRepositoryServices<OrderItem, string>
+    {
+        private readonly RestaurantReservationDbContext _dbContext;
+
+        public OrderItemRepository(RestaurantReservationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<OperationResult<string>> AddAsync(OrderItem orderItem)
+        {
+            try
+            {
+                await _dbContext.OrderItems.AddAsync(orderItem);
+                await _dbContext.SaveChangesAsync();
+                return OperationResult<string>.SuccessResult($"id:{orderItem.Id}");
+            }
+            catch (DbUpdateException ex) when (IsForeignKeyViolation(ex))
+            {
+                return OperationResult<string>.FailureResult("Foreign key constraint violation. Make sure associated entities exist.");
+            }
+            catch (Exception ex)
+            {
+                return OperationResult<string>.FailureResult($"An error occurred while processing the reservation.{ex.Message}");
+            }
+            bool IsForeignKeyViolation(DbUpdateException ex)
+            {
+                var sqlException = ex.InnerException as SqlException;
+
+                return sqlException?.Number == 547;
```

Comment:
Magic number, but I would say it might be fine in this small context. will explain